Common/Scheduling/TimeMonitor.cs

```diff
@@ -26,7 +26,7 @@ namespace QuantConnect.Scheduling
     /// </summary>
     public class TimeMonitor : IDisposable
     {
-        private readonly List<TimeConsumer> _timeConsumers;
+        protected readonly List<TimeConsumer> _timeConsumers;
```

Comment:
nit - `_timeConsumers`  -> `TimeConsumers` would be nice to add the xml documentation and a remarks section mentioning it's protected for testing, same for `ProcessConsumer`